float32 grid_size
int32 resolution
---
bool done
